src/Database/Database.php

```diff
@@ -2791,24 +2791,37 @@ public function updateDocument(string $collection, string $id, Document $documen
         }
 
         $time = DateTime::now();
-        $document->setAttribute('$updatedAt', $time);
-
         $old = Authorization::skip(fn () => $this->silent(fn () => $this->getDocument($collection, $id))); // Skip ensures user does not need read permission for this
+
         $collection = $this->silent(fn () => $this->getCollection($collection));
 
         $validator = new Authorization(self::PERMISSION_UPDATE);
 
         if ($collection->getId() !== self::METADATA) {
             $documentSecurity = $collection->getAttribute('documentSecurity', false);
 
-            if (!$validator->isValid([
+            $skipPermission = true;
+            //compare if the document any changes
```

Comment:
Let's be consistent with the style:

```suggestion
            // Compare if the document has any changes
```